### ISSUE TITLE
False negative non-overlapping equality check on uuid

### ISSUE THREAD

**Issue by Ha-limLee** [python/mypy#21627]
**Bug Report**

In strict mode, if I compare `UUID` variable to other typed one(e.g., `int`), I get no error.

**To Reproduce**

```python
import uuid

a: uuid.UUID = uuid.uuid4()
b: int = 1

a == b
```

see: https://mypy-play.net/?mypy=2.1.0&python=3.12&flags=strict&gist=8c00e51d1127d13a1c52d2172526e157

**Expected Behavior**

error: Non-overlapping equality check (left operand type: "UUID", right operand type: "int")  [comparison-overlap]

**Actual Behavior**

no error.

**Your Environment**

- Mypy version used: 2.1.0
- Mypy command-line flags: strict
- Python version used: 3.12

**Comment by ilevkivskyi** [comment]
Cannot reproduce this.

**Comment by Ha-limLee** [comment]
Hi @ilevkivskyi. thank you for quick response!

Oh my bad. I added wrong playground URL. I edited the link to right one.

Here's playground link: https://mypy-play.net/?mypy=2.1.0&python=3.12&flags=strict&gist=8c00e51d1127d13a1c52d2172526e157

Please check again.
